OPHD/Map/MapOffset.h

```diff
@@ -8,3 +8,41 @@ struct MapOffset
 	NAS2D::Vector<int> xy;
 	int z;
 };
+
+
+inline bool operator==(const MapOffset& offset1, const MapOffset& offset2)
+{
+	return offset1.xy == offset2.xy && offset1.z == offset2.z;
+}
+
+
+inline bool operator!=(const MapOffset& offset1, const MapOffset& offset2)
+{
+	return offset1.xy != offset2.xy || offset1.z != offset2.z;
+}
+
+
+inline MapOffset operator*(const MapOffset& offset, int scalar)
+{
+	MapOffset result;
+	result.xy = offset.xy * scalar;
+	result.z = offset.z * scalar;
+	return result;
+}
```

Comment:
When you also support `operator*=`, you can use that to implement `operator*`. Simply remove the `const &` from the parameter passing so it generates a copy, and then use `operator*=` on the copy.

```cpp
inline MapOffset operator*(MapOffset offset, int scalar)
{
	return offset *= scalar;
}
```